lib/msal-browser/src/cache/BrowserStorage.ts

```diff
@@ -271,4 +271,102 @@ export class BrowserStorage implements ICacheStorage {
+     * @param serverAuthenticationRequest
+     * @param account
+     */
+    updateCacheEntries(state: string, nonce: string, authorityInstance: string): void {
+        // Cache the request state
+        this.setItem(TemporaryCacheKeys.REQUEST_STATE, state);
+
+        // Cache the nonce
+        this.setItem(this.generateNonceKey(state), nonce);
+
+        // Cache authorityKey
+        this.setAuthorityCache(authorityInstance, state);
+    }
+
+    /**
+     * Reset all temporary cache items
+     * @param state
+     */
+    resetRequestCache(state: string): void {
+        // check state and remove associated cache items
+        this.getKeys().forEach(key => {
+            if (!StringUtils.isEmpty(state) && key.indexOf(state) !== -1) {
+                const splitKey = key.split(Constants.RESOURCE_DELIM);
+                const keyState = splitKey.length > 1 ? splitKey[splitKey.length-1]: null;
+                if (keyState === state) {
+                    this.removeItem(key);
+                }
+            }
+        });
+
+        // delete generic interactive request parameters
+        this.removeItem(TemporaryCacheKeys.REQUEST_STATE);
+        this.removeItem(TemporaryCacheKeys.REQUEST_PARAMS);
+        this.removeItem(TemporaryCacheKeys.ORIGIN_URI);
+    }
+
+    cleanRequest(): void {
+        // Interaction is completed - remove interaction status.
+        this.removeItem(BrowserConstants.INTERACTION_STATUS_KEY);
+        const cachedState = this.getItem(TemporaryCacheKeys.REQUEST_STATE);
+        this.resetRequestCache(cachedState || "");
+        this.removeItem(TemporaryCacheKeys.REQUEST_STATE);
+    }
+
+    cacheCodeRequest(authCodeRequest: AuthorizationCodeRequest, browserCrypto: ICrypto): void {
+        this.setItem(TemporaryCacheKeys.REQUEST_PARAMS, browserCrypto.base64Encode(JSON.stringify(authCodeRequest)));
+    }
+
+    /**
+     * Gets the token exchange parameters from the cache. Throws an error if nothing is found.
+     */
+    getCachedRequest(state: string, browserCrypto: ICrypto): AuthorizationCodeRequest {
+        try {
+            // Get token request from cache and parse as TokenExchangeParameters.
+            const encodedTokenRequest = this.getItem(TemporaryCacheKeys.REQUEST_PARAMS);
+            const parsedRequest = JSON.parse(browserCrypto.base64Decode(encodedTokenRequest)) as AuthorizationCodeRequest;
+            this.removeItem(TemporaryCacheKeys.REQUEST_PARAMS);
+            // Get cached authority and use if no authority is cached with request.
+            if (StringUtils.isEmpty(parsedRequest.authority)) {
+                const authorityKey: string = this.generateAuthorityKey(state);
+                const cachedAuthority: string = this.getItem(authorityKey);
+                parsedRequest.authority = cachedAuthority;
+            }
+            return parsedRequest;
+        } catch (err) {
+            throw BrowserAuthError.createTokenRequestCacheError(err);
+        }
+    }
```

Comment:
+1 for this. All browser specific cache handling now is clearly demarcated between `msal-common` and `msal-browser`